content/documentation/staging/configuration/health/index.adoc

```diff
@@ -0,0 +1,384 @@
+---
+title: "Health Configuration"
+date: 2019-03-06T13:55:38+02:00
+draft: false
+type: "documentation"
+weight: 5
+---
+
+:linkattrs:
+:sectlinks:
+
+= Health Configuration
+:sectnums:
+:toc: left
+toc::[]
+:toc-title: Custom Health
+:keywords: Kiali Documentation Health
+:icons: font
+:imagesdir: /images/documentation/health-configuration/
+
+
+== Health
+
+Customize resource health by overriding the default health configuration.
+
+
+=== Icons and colors
+
+Kiali use icons and colors to indicate the health of resources and associated request traffic.
+
+++++
+<div style="display: flex;">
+<ul>
+<li>
+ <img src="/images/documentation/health-configuration/no_health.png" style="width: 40px;height: 40px" /> No Health Information (NA)
+</li>
+<li>
+ <img src="/images/documentation/health-configuration/healthy.png" style="width: 40px;height: 40px" /> Healthy
+</li>
+<li>
+ <img src="/images/documentation/health-configuration/degraded.png" style="width: 40px;height: 40px" /> Degraded
+</li>
+<li>
+ <img src="/images/documentation/health-configuration/failure.png" style="width: 40px;height: 40px" /> Failure
+</li>
+</ul>
+  </div>
+++++
+
+=== Default Values
+
+==== Request Traffic
+
+By default Kiali uses the traffic rate configuration shown below.  Application errors have minimal tolerance while client errors have a higher tolerance reflecting that some level of client errors is often normal (e.g. 404 Not Found):
+
+* For _http_ protocol 4xx are client errors and 5xx codes are application errors.
+* For _grpc_ protocol all 1-16 are errors (0 is success).
+
+So, for example, if the rate of application errors is >= 0.1% kiali will show `Degraded` health and if > 10% will show `Failure` health.
+
+```yaml
+# ...
+  health_config:
+    rate:
+      - namespace: ".*"
+        kind: ".*"
+        name: ".*"
+        tolerance:
+          - code: "^5\\d\\d$"
+            direction: ".*"
+            protocol: "http"
+            degraded: 0.1
```

Comment:
Why don't leave this to 0 ?
To have a 0.1 is really odd to explain.
Perhaps I missed this in the code, but I suggest to adjust this.

Reply:
It was in 0.1 % for me is the same to set it to 0 or let it....In the code I just checking if error_rate is > 0 and > degraded value so it's going to work with 0. It backend we  have this values so we should change them if we want to keep this aligned